parsers/regex/regex.go

```diff
@@ -0,0 +1,158 @@
+// Package regex consumes logs given user-defined regex format for lines
+
+/* Example usage (from honeytail root)
+  		go install . && \
+ 			honeytail -p regex -k $HONEYTAIL_WRITEKEY \
+			-f /private/var/system.log \
+			--dataset 'MY_TEST_DATASET' \
+			--backfill \
+			--regex.line_regex="(?P<time>\d{2}:\d{2}:\d{2})" \
+			--regex.timefield="time" \
+			--regex.time_format="%H:%M:%S"
+*/
+
+package regex
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/honeycombio/honeytail/event"
+	"github.com/honeycombio/honeytail/httime"
+	"github.com/honeycombio/honeytail/parsers"
+)
+
+type Options struct {
+	LineRegex       string `long:"line_regex" description:"a regular expression with named capture groups representing the fields you want parsed"`
```

Comment:
Let's make sure to add some documentation and examples to the website, and link to it from this help string. Capture group syntax varies between languages, so this could be confusing otherwise.